### PR TITLE
CDK Example

### PR DESCRIPTION
**Pull request by turlockmike**
Added an aws cdk example using the SAR to deploy the layer. There might be an easier way to import the layer into the CDK lambda construct, but not sure how.

Tested the example, should work as long as you install cdk globally.

### CODE REVIEW

File: example-aws-cdk/cdk/example-aws-cdk-stack.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+export class ExampleAwsCdkStack extends cdk.Stack {
+    constructor(scope: cdk.App, id: string, props?: cdk.StackProps) {
+        super(scope, id, props);
+
+
+        const denoRuntime = new CfnApplication(this, 'DenoRuntime', {
+            location: {
+                applicationId: 'arn:aws:serverlessrepo:us-east-1:390065572566:applications/deno',
+                semanticVersion: '1.1.1'
```
Comment (hayd) [inline example-aws-cdk/cdk/example-aws-cdk-stack.ts:14]: neat

**Comment by hayd** [comment]
I'd quite like to add a test for this (not necessarily in this PR).
Can cdk run a local test of a deploy? (Failing that we can cdk deploy in the [publish like for sam](https://github.com/hayd/deno-lambda/blob/806b65ffcf8b89a4969696c69b6cf79b7e3fc2ec/.github/workflows/publish.yml#L68).)

**Comment by turlockmike** [comment]
You can use the `cdk synth` command to generate a cloud formation template and write a snapshot test to ensure that it produces the right output. 

The `cdk deploy` command essentially does 2 things. 
1. Write the code assets to an s3 bucket
2. Push the cloud formation template up to the aws account using the local aws cli. 

The other option is, if you have is you can deploy this to whatever AWS environment you want to during CI and just run a test to make sure the endpoint responds as expected. The endpoint is part of the output of the deploy function for coud formation. 
 That'd basically be a full e2e test, but it would cover pretty much everything.

**Comment by hayd** [comment]
`cdk docker` may also be an option. xlink: https://docs.aws.amazon.com/cdk/latest/guide/tools.html

Thanks again!

File: example-aws-cdk/yarn.lock [hunk 2]
```diff
@@ -0,0 +1,1606 @@
...
+  version "1.0.0"
+  resolved "https://registry.yarnpkg.com/astral-regex/-/astral-regex-1.0.0.tgz#6c8c3fb827dd43ee3918f27b82782ab7658a6fd9"
+  integrity sha512-+Ryf6g3BKoRc7jfp7ad8tM4TtMiaWvbF/1/sQcZPkkS7ag3D5nMBCe2UfOTONtAkaG0tO0ij3C5Lwmf1EiyjHg==
+
+async@^2.6.3:
+  version "2.6.3"
+  resolved "https://registry.yarnpkg.com/async/-/async-2.6.3.tgz#d72625e2344a3656e3a3ad4fa749fa83299d82ff"
+  integrity sha512-zflvls11DCy+dQWzTW2dzuilv8Z5X/pjfmZOWba6TNIVDm+2UDaJmXSOXlasHKfNBs8oo3M0aT50fDEWfKZjXg==
+  dependencies:
+    lodash "^4.17.14"
```
Comment (hayd) [inline example-aws-cdk/yarn.lock:372]: I am getting an github security alert about lodash vuln (<4.17.19).

Presumably I can update to ^4.17.19 ... but could we get away removing yarn.lock entirely? (or is it important to commit it?)

**Pull request merged**